tests/D2L.CodeStyle.Analyzers.Test/Async/Generator/SyncGeneratorTests.cs

```diff
@@ -0,0 +1,180 @@
+﻿using D2L.CodeStyle.Analyzers.Async.Generator;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using NUnit.Framework;
+
+namespace D2L.CodeStyle.Analyzers.Tests.Async.Generator;
+
+/// <summary>
+/// Integration tests for the generator as a whole.
+/// Prefer unit tests of AsyncToSyncTransformer and FileCollector as they are
+/// simpler. We just want a bit of smoke-testing here.
+/// </summary>

```

Comment:
I would change my tune on this if we had something as good as spec tests for generators, but I'm not feeling the urge to rush into that right now personally.